### DISCUSSION TITLE
Add the capability to sample trajectories

### DISCUSSION THREAD

**Discussion by jamartinh** [ymd-h/cpprb#31] [Feature Request]
For some algorithms it is very useful to be able to sample random trajectories of max_len to take advantage of the temporal series information.

For instance, rb.sample_trajectories(n_samples: int, trajectory_max_len: int)

So you get a collection of samples each one containing an ordered part of an experienced episode

**Comment by ymd-h** [comment]
@jamartinh 

Thank you for your feedback.

Since we couldn't define agreeable standard usage, we don't provide a specific class for episodic experiment (at least now),
however, I still believe we can build an episodic configuration with current `(Prioritized)ReplayBuffer`
by setting a trajectory (aka. episode) as a single record (= step in non-episodic experiment).


If the trajectory is always same length (= `T`), the situation becomes simple.
At the `env_dict` parameter of the constructor, you can use `(T, *original_shape)` or `(*original_shape, T)` for the episodic values.
Then you can `add` whole trajectory at once.
If you need temporary storage for ongoing current episode, you might use another `ReplayBuffer`.


If the trajectory length differ each time, I think there are 2 options.

1st option: Use padding
As far as I know, anyway we pad data before passing neural network, so that it makes sense for me to pad trajectory before storing.
This might requires unnecessary memory.

2nd option: Use `dtype=object`
By specifying `dtype` as `object` (with `shape` as `1`) , Python object are stored.
Python object is a kind of pointer, so that it is also possible someone might change the actual data.


I hope it will help you.
